kubechain/internal/controller/taskruntoolcall/taskruntoolcall_controller.go

```diff
@@ -732,6 +857,16 @@ func (r *TaskRunToolCallReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		r.MCPManager = mcpmanager.NewMCPServerManager()
 	}
 
+	if r.HLClient == nil {
```

Comment:
A note, I'm piggybacking here and probably breaking the naming of this function, maybe worth looking at renaming?